### ISSUE TITLE
Vhost as a Path parameter for http auth backend 

### ISSUE THREAD

**Issue by devansharora68** [rabbitmq/rabbitmq-auth-backend-http#97]
**Currently**  http Auth backend supports GET and POST REST call types 
For **GET** type request, vhost is a part of query params 
Like below 
{user_path,   "https://ahost1:8080/auth/user?vhost=vhost01&username=user01&password=1234"}
{vhost_path,  "https://ahost1:8080/auth/vhost?vhost=vhost01&username=user01}

For **POST** type request, vhost is part of request body.

**We want support for vhost to be a PATH param in GET and POST requests** 
Like below 
{user_path,     "https://ahost1:8080/auth/user/{vhost}"}
{vhost_path,    "https://ahost1:8080/auth/vhost/{vhost}"}

So that we are able to define url path based routing in application gateways and httpd servers.

As vhost is a specific resource, then can it be a Path param.
As discussed here 
[https://github.com/rabbitmq/rabbitmq-auth-backend-http/issues/96](https://github.com/rabbitmq/rabbitmq-auth-backend-http/issues/96)

**Comment by acogoluegnes** [comment]
The `{vhost}` path variable could be in every endpoint (`user_path`, `vhost_path`, `resource_path`, and `topic_path`)?

Note the virtual host is not available in the `user_path` call, so it won't be possible to have it as a path variable for this specific URI. Does that make sense?

**Comment by michaelklishin** [comment]
@acogoluegnes it does to me :)

**Comment by devansharora68** [comment]
@acogoluegnes 
Yes {vhost} path variable should in every endpoint.

Also, We have observed vhost comes as kind of optional query param in user_path call as well. 
I checked and it sends vhost value as well like when a authentication call is from web_mqtt plugin.
My current service signature for user_path backend is :

```
public ResponseEntity<?> user(@RequestParam("username") String username,
			@RequestParam("password") String password,@RequestParam(value="vhost",required = false) String vhost)
```

**Comment by acogoluegnes** [comment]
The virtual host is propagated for protocols like STOMP and MQTT, but not for AMQP. Let me see if there's something we can do about it.

**Comment by michaelklishin** [comment]
`user_path` is an authentication endpoint. Why would anyone need a virtual host as an argument there?

**Comment by acogoluegnes** [comment]
The virtual host cannot be propagated to `user_path` for AMQP connections. The fact it's propagated for protocols like STOMP and MQTT is just a lucky circumstance.

**Comment by devansharora68** [comment]
Agreed for user_path @michaelklishin  @acogoluegnes 

Would need {vhost} in services other than user_path.
Still if optionally mqtt & stomp can propagate no harm for us i guess.

**Comment by acogoluegnes** [comment]
Yes, the propagation should not change, it's just the `{vhost}` replacement won't be available for `user_path` for AMQP connections.

**Comment by devansharora68** [comment]
@acogoluegnes Hey Arnaud, what is the process now to achieve this change?

**Comment by acogoluegnes** [comment]
We'll add this task to our backlog. We provide no ETA. A PR is always welcome and will be reviewed.

**Comment by devansharora68** [comment]
OK @acogoluegnes . How would you suggest us doing this change? your guidance will help :)

**Comment by michaelklishin** [comment]
@devansharora68 LDAP is one implementation of the [authZ interface](https://github.com/rabbitmq/rabbitmq-common/blob/master/src/rabbit_authz_backend.erl) that [supports variables](https://github.com/rabbitmq/rabbitmq-auth-backend-ldap/blob/master/src/rabbit_auth_backend_ldap.erl#L115) (and here are the [end user docs](https://www.rabbitmq.com/ldap.html#query-types)). [rabbit_auth_backend_http](https://github.com/rabbitmq/rabbitmq-auth-backend-http/blob/master/src/rabbit_auth_backend_http.erl) is the module that implements it in this plugin.

**Comment by devansharora68** [comment]
Thanks @michaelklishin
